File: other/sorts/cocktail.rb
def cocktail array
    array.count.times do |pass|
        if pass.even?
            array.count.times do |i|
                if foo = array[i + 1]
                    bar = array[i]
                    if foo < bar
                        array[i] = foo
                        array[i + 1] = bar
                    end
                end
            end
        else
            (1..array.count).each do |i|
                if foo = array[-i - 1]
                    bar = array[-i]
                    if foo > bar
                        array[-i - 1] = bar
                        array[-i] = foo
                    end
                end
            end
        end
    end
    array
end

File: other/sorts/cocktail.js
function cocktail(array){
    for (var x = 0; x < array.length; x++){
        for (var y = 0; y < array.length - 1; y++){
            if (x % 2 == 0){
                var foo = array[y]
                var bar = array[y + 1]
                if (foo > bar) {
                    array[y] = bar
                    array[y + 1] = foo
                }
            } else {
                var foo = array[-y]
                var bar = array[-y - 1]
                if (foo < bar) {
                    array[y] = bar
                    array[y + 1] = foo
                }
            }
        }
    }
    return array
}